src/test/java/core/basesyntax/RegistrationServiceTest.java

```diff
@@ -0,0 +1,116 @@
+package core.basesyntax;
+
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exception.RegistrationException;
+import core.basesyntax.model.User;
+import core.basesyntax.service.RegistrationServiceImpl;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+public class RegistrationServiceTest {
+    private static final String CORRECT_LOGIN = "Valid User Login";
+    private static final String IDENTICAL_LOGIN = "Identical User Login";
+    private static final String CORRECT_PASSWORD = "Valid User Password";
+    private static final String INVALID_LENGTH_5 = "55555";
+    private static final Integer CORRECT_AGE = 20;
+
+    private StorageDaoImpl storageDao;
+    private RegistrationServiceImpl registrationService;
+
+    @BeforeEach
+    void setUp() {
+        storageDao = new StorageDaoImpl();
+        registrationService = new RegistrationServiceImpl();
+    }
+
+    @Test
+    public void register_nullUser_notOk() {
+        assertThrows(RegistrationException.class, () -> registrationService.register(null),
+                "User should not be null");
+    }
+
+    @Test
+    public void register_validUser_ok() {
+        User validUser = new User();
+        validUser.setLogin(CORRECT_LOGIN);
+        validUser.setPassword(CORRECT_PASSWORD);
+        validUser.setAge(CORRECT_AGE);
+
+        registrationService.register(validUser);
+        assertNotNull(storageDao.get(CORRECT_LOGIN));
+    }
+
+    @Test
+    public void register_invalidLogin_notOk() {
+        User validUser = new User();
+        validUser.setLogin(IDENTICAL_LOGIN);
```

Comment:
why is it a valid user?

Reply:
because this user was registered first with a unique login and the second user has the same login, that's why second user is invalid.